### PR TITLE
Wrap all select expressions in aliases with their original string expression

### PR DESCRIPTION
**Pull request by zachmu**
This makes the result set schema exactly match the select expressions in the query.

### CODE REVIEW

**Review by max-hoffman — APPROVED** [review]
core of the change looks good, i'm not sure how some of the testing/having/greater code is related

File: enginetest/query_plans.go [hunk 1]
```diff
@@ -38,24 +38,26 @@ var PlanTests = []QueryPlanTest{
 	{
 		Query: `select row_number() over (order by i desc), mytable.i as i2 
 				from mytable join othertable on i = i2 order by 1`,
-		ExpectedPlan: "Sort(row_number() over ( order by [mytable.i, idx=0, type=BIGINT, nullable=false] DESC) ASC)\n" +
-			" └─ Window(row_number() over ( order by [mytable.i, idx=0, type=BIGINT, nullable=false] DESC), mytable.i as i2)\n" +
-			"     └─ IndexedJoin(mytable.i = othertable.i2)\n" +
-			"         ├─ Table(mytable)\n" +
-			"         └─ IndexedTableAccess(othertable on [othertable.i2])\n" +
+		ExpectedPlan: "Sort(row_number() over (order by i desc) ASC)\n" +
+			" └─ Project(row_number() over ( order by [mytable.i, idx=0, type=BIGINT, nullable=false] DESC) as row_number() over (order by i desc), i2)\n" +
```
[the remark this answers is not in the document]
Comment (zachmu) [reply enginetest/query_plans.go:42]: Right, explicit aliases are left alone.

File: enginetest/queries.go [hunk 2]
```diff
@@ -1813,7 +1813,39 @@ var QueryTests = []QueryTest{
 		},
 	},
 	{
-		Query: "SELECT i, SUM(i) FROM mytable GROUP BY i ORDER BY SUM(i) DESC",
+		Query: "SELECT SUM(i) as sum, i FROM mytable GROUP BY i ORDER BY 1+SUM(i) ASC",
```
[the remark this answers is not in the document]
Comment (zachmu) [reply enginetest/queries.go:1816]: They don't... when I introduced these aliases it exposed bugs in other parts of the analyzer. So I wrote a bunch more query tests related to the things that broke.

File: sql/expression/function/greatest_least.go [hunk 3]
```diff
@@ -215,7 +210,12 @@ func (f *Greatest) FunctionName() string {
 }
 
 // Type implements the Expression interface.
-func (f *Greatest) Type() sql.Type { return f.returnType }
+func (f *Greatest) Type() sql.Type {
```
[the remark this answers is not in the document]
Comment (zachmu) [reply sql/expression/function/greatest_least.go:213]: Yes, these two functions relied on the arguments having already been resolved when the function was created. Other functions were ok... I think. I'd better do an audit now that you bring this up.

**Pull request merged**